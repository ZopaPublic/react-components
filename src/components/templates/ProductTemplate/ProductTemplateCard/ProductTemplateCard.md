### Summary

`<ProductTemplateCard>` is a wrapper against the `<Card />` component with fixed styling for consistency between products.
