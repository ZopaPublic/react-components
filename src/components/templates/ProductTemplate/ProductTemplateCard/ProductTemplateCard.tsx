import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';

import { colors, grid } from '../../../../constants';
import Card from '../../../organisms/Card/Card/Card';

interface ProductTemplateCardProps {
  className?: string;
  children: React.ReactNode;
}

const StyledCard = styled(Card)`
  @media (max-width: ${grid.breakpoints.m}px) {
    border-radius: 0;
    box-shadow: 0 0 1px 0 ${colors.greyLight};
    border: none;
    border-bottom: 1px solid ${colors.greyLighter};
  }
`;

export function ProductTemplateCard({ children, className }: ProductTemplateCardProps) {
  return <StyledCard className={classnames('px-4 py-6 m:p-8', className || '')}>{children}</StyledCard>;
}
