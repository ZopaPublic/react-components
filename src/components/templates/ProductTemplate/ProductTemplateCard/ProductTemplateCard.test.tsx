import React from 'react';
import { render } from '@testing-library/react';
import { ProductTemplateCard } from './ProductTemplateCard';

describe('<ProductTemplateCard />', () => {
  it('renders with all the props', () => {
    const { container } = render(<ProductTemplateCard>All the children</ProductTemplateCard>);
    expect(container).toMatchSnapshot();
  });
});
