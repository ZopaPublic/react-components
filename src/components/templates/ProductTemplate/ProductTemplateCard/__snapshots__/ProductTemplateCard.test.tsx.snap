// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTemplateCard /> renders with all the props 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  background-color: #FFFFFF;
  border: 1px solid #EFEFEF;
  border-radius: 12px;
  box-shadow: 0 1px 0 0 #D4D7D9;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > .c2 {
  -webkit-flex: 0 1 200px;
  -ms-flex: 0 1 200px;
  flex: 0 1 200px;
}

.c0 .zrc__card-heading {
  font-size: 16px;
}

@media (max-width:768px) {
  .c1 {
    border-radius: 0;
    box-shadow: 0 0 1px 0 #D4D7D9;
    border: none;
    border-bottom: 1px solid #EFEFEF;
  }
}

<div>
  <div
    class="c0 c1 px-4 py-6 m:p-8"
  >
    All the children
  </div>
</div>
`;
